### ISSUE TITLE
provide project method

### ISSUE THREAD

**Issue by goldingn** [greta-dev/greta#21]
Provide a `project` so that after sampling from (or optimising) a model, users can add extra nodes to the dag and project the draws to them:

```r
draws <- mcmc(m, ...)
mu_new <- alpha + X_new %*% beta
abs_beta <- abs(beta)
draws_mu_new <- project(draws, mu_new, abs_beta)
```

This depends on #20

**Comment by goldingn** [comment]
this would error if `draws` didn't trace all of the required variables

**Comment by goldingn** [comment]
This would be a thin wrapper around the solution to #52

**Comment by goldingn** [comment]
(now enabled as `evaluate()` in `goldingn/greta@dev`)

**Comment by goldingn** [comment]
Switched to `calculate()` and released in v0.2.4
